my_game.py

```diff
@@ -165,21 +169,31 @@ def on_update(self, delta_time):
         Movement and game logic
         """
 
-        # Calculate player speed based on the keys pressed
-        self.player.change_x = 0
-
-        # Move player with keyboard
+        # Set x/y speed for player with arrow keys
+        # Left/right
         if self.left_pressed and not self.right_pressed:
-            self.player.change_x = -PLAYER_SPEED_X
+            self.player.change_x = -PLAYER_SPEED
         elif self.right_pressed and not self.left_pressed:
-            self.player.change_x = PLAYER_SPEED_X
+            self.player.change_x = PLAYER_SPEED
+        else:
+            self.player.change_x = 0
+
+        # Up/down
+        if self.up_pressed and not self.down_pressed:
+            self.player.change_y = PLAYER_SPEED
```

Comment:
Currently, change_x and change_y can equal PLAYER_SPEED - this means we move much faster diagonally.
I think the _sum of the two_ should equal PLAYER_SPEED when we are moving.

Reply:
I have reverted back to movement only in x and y. A later PR can add diagonal movement. This PR is about the physics engine.